scripts/bootstrap.sh.in

```diff
@@ -261,7 +261,7 @@ orchestrate_bootstrap() {
 
     run "Deploying early-stage bootstrap node in local mode (this may take a while)" \
         "${SALT_CALL}" --local --state-output=mixed --retcode-passthrough state.sls \
-            '["metalk8s.roles.bootstrap", "metalk8s.roles.minion"]' \
+            '["metalk8s.kubernetes.ca.kubernetes.installed", "metalk8s.roles.bootstrap", "metalk8s.roles.minion"]' \
```

Comment:
I think it makes sense to only keep `metalk8s.roles.*` SLSs here.

Instead, couldn't the state in which we create a config file which points to this key/cert pair 'include' the relevant state?

Reply:
The problem is that we generate these key/cert here because we need them to create the salt-api certificate but we not use the CA key/cert 
If we want to include the state somewhere it will likely be in the salt-master state or in the roles.bootstrap state.
Personnaly don't really like it, that mean bootstrap roles will also deploy the kubernetes root CA. Which is not what we want IMO

Reply:
Well, we *do* use that CA's key/cert, *exactly* to generate the SaltAPI cert.

- `metalk8s.roles.bootstrap` requires the CA to be available, otherwise we can't sign the SaltAPI cert it wants/needs to deploy
- Implicitly, this means `metalk8s.roles.bootstrap` (or a state it contains, e.g. `salt-master`) requires the `ca` minion to be provisioned
- We're in bootstrap phase of the cluster, so `ca` minion == bootstrap minion (there's only one node...)
- Ergo, how about simply adding `metalk8s.roles.ca` in this list, before `metalk8s.roles.bootstrap`?

Reply:
> * Ergo, how about simply adding `metalk8s.roles.ca` in this list, before `metalk8s.roles.bootstrap`?

Yes we can but we need to rewrite all the CA states because for the moment Ca states also push the certificate to the mine but if we have no salt-master we can't push to the mine. (At least that does not fail but nothing is really pushed, so maybe we can do that but we will need to run the state again after the salt-master deployment)

> * `metalk8s.roles.bootstrap` requires the CA to be available, otherwise we can't sign the SaltAPI cert it wants/needs to deploy
> * Implicitly, this means `metalk8s.roles.bootstrap` (or a state it contains, e.g. `salt-master`) requires the `ca` minion to be provisioned

Yes I agree, but the salt states are not used only for the first deployment compare to the bootstrap script so for me it's better to have this behavior only in the script so that we don't have a requirement in our salt formulas to use the same node for `ca` and `salt-master`

Reply:
OK